Self-Consistent-Field/RHF_symmetry.py

```diff
@@ -0,0 +1,192 @@
+"""A restricted Hartree-Fock script using the Psi4NumPy formalism that
+accounts for molecular symmetry using symmetrized orbitals.
+
+References:
+- Algorithm taken from [Szabo:1996], pp. 146
+- Equations taken from [Szabo:1996]
+"""
+
+__authors__ = "Eric J. Berquist"
+__credits__ = ["Eric J. Berquist", "Daniel G. A. Smith"]
+
+__copyright__ = "(c) 2014-2018, The Psi4NumPy Developers"
+__license__ = "BSD-3-Clause"
+__date__ = "2018-12-26"
+
+import time
+import numpy as np
+np.set_printoptions(precision=8, linewidth=200, suppress=True)
+import psi4
+from helper_HF import transform_aotoso, transform_sotoao
+
+# Memory for Psi4 in GB
+psi4.set_memory('500 MB')
+psi4.core.set_output_file("output.dat", False)
+
+# Memory for NumPy in GB
+numpy_memory = 2
+
+mol = psi4.geometry("""
+O
+H 1 1.1
+H 1 1.1 2 104
+""")
+
+psi4.set_options({'basis': 'sto-3g',
+                  'scf_type': 'direct',
+                  'e_convergence': 1e-8})
+
+# Set defaults
+maxiter = 50
+E_conv = 1.0E-8
+D_conv = 1.0E-7
+
+# Integral generation from Psi4's MintsHelper, which automatically
+# performs symmetry adaptation.
+wfn = psi4.core.Wavefunction.build(mol, psi4.core.get_global_option('BASIS'))
+t = time.time()
+mints = psi4.core.MintsHelper(wfn.basisset())
+
+nirrep = wfn.nirrep()
+# Water belongs to the C2v point group, which has four irreducable
+# representations (irreps): A_1, A_2, B_1, and B_2.
+assert nirrep == 4
+nsopi = list(wfn.nsopi())
+
+# Get nbf and ndocc for closed shell molecules
+nbf = sum(nsopi)
+ndocc = wfn.nalpha()
+
+print('\nNumber of occupied orbitals: %d' % ndocc)
+print('Number of basis functions: %d' % nbf)
+print('Number of spin orbitals per irrep:', nsopi)
+
+# Run a quick check to make sure everything will fit into memory
+I_Size = (nbf**4) * 8.e-9
+print("\nSize of the ERI tensor will be %4.2f GB." % I_Size)
+
+# Estimate memory usage
+memory_footprint = I_Size * 1.5
+if I_Size > numpy_memory:
+    psi4.core.clean()
+    raise Exception("Estimated memory utilization (%4.2f GB) exceeds numpy_memory "
+                    "limit of %4.2f GB." % (memory_footprint, numpy_memory))
+
+# A matrix is returned for every irrep, even if the irrep is not present for
+# the molecule; these must be filtered out to avoid problems with NumPy arrays
+# that have a zero dimension.
+def filter_empty_irrep(coll):
+    return tuple(m for m in coll if all(m.shape))
+
+# The convention will be to have quantities in the spin orbital (SO) basis
+# ending with an underscore, each consisting of one matrix per irrep.
+S_ = filter_empty_irrep(mints.so_overlap().to_array())
+T_ = filter_empty_irrep(mints.so_kinetic().to_array())
+V_ = filter_empty_irrep(mints.so_potential().to_array())
+# The two-electron integrals are not blocked according to symmetry, so a
+# transformation between the atomic orbital (AO) and SO bases will be required.
+I_AO = np.asarray(mints.ao_eri())
```

Comment:
Small thing, but can you add a blank line between the code and new comment blocks?